src/main/java/de/zalando/tip/zalenium/proxy/DockerSeleniumStarterRemoteProxy.java

```diff
@@ -118,36 +112,65 @@ private static void readConfigurationFromEnvVariables() {
 
         String tz = env.getStringEnvVariable(ZALENIUM_TZ, DEFAULT_TZ);
         setTimeZone(tz);
+
+        String containerN = env.getStringEnvVariable(ZALENIUM_CONTAINER_NAME, DEFAULT_ZALENIUM_CONTAINER_NAME);
+        setContainerName(containerN);
     }
 
     /*
      *  Updating the proxy's registration request information with the current DockerSelenium capabilities.
      *  If it is not possible to retrieve them, then we default to Chrome and Firefox in Linux.
      */
     @VisibleForTesting
-    static RegistrationRequest updateDSCapabilities(RegistrationRequest registrationRequest, String url) {
+    protected static RegistrationRequest updateDSCapabilities(RegistrationRequest registrationRequest) {
         registrationRequest.getCapabilities().clear();
-        registrationRequest.getCapabilities().addAll(getCapabilities(url));
+        registrationRequest.getCapabilities().addAll(getCapabilities());
         return registrationRequest;
     }
 
+    @SuppressWarnings("ConstantConditions")
     @VisibleForTesting
-    static void clearCapabilities() {
-        dockerSeleniumCapabilities.clear();
-    }
-
-    private static List<DesiredCapabilities> getCapabilities(String url) {
+    public static List<DesiredCapabilities> getCapabilities() {
         if (!dockerSeleniumCapabilities.isEmpty()) {
             return dockerSeleniumCapabilities;
         }
 
-        dockerSeleniumCapabilities = getDockerSeleniumCapabilitiesFromGitHub(url);
-        if (dockerSeleniumCapabilities.isEmpty()) {
-            dockerSeleniumCapabilities = getDockerSeleniumFallbackCapabilities();
-            LOGGER.log(Level.WARNING, LOGGING_PREFIX + "Could not fetch capabilities from {0}, falling back to defaults.", url);
-            return dockerSeleniumCapabilities;
+        String chromeVersion = "";
+        String firefoxVersion = "";
+
+        // Getting versions from the current docker-selenium image
+        try {
+            String latestDownloadedImage = getLatestDownloadedImage(DOCKER_SELENIUM_IMAGE);
+            ImageInfo imageInfo = dockerClient.inspectImage(latestDownloadedImage);
+            chromeVersion = imageInfo.config().labels().get("selenium2_chrome_version");
+            firefoxVersion = imageInfo.config().labels().get("selenium2_firefox_version");
+        } catch (DockerException | InterruptedException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)